command/agent/job_endpoint.go

```diff
@@ -1809,11 +1805,23 @@ func apiLogConfigToStructs(in *api.LogConfig) *structs.LogConfig {
 	if in == nil {
 		return nil
 	}
-	return &structs.LogConfig{
-		Enabled:       *in.Enabled,
+	out := &structs.LogConfig{
+		Disabled:      dereferenceBool(in.Disabled),
 		MaxFiles:      dereferenceInt(in.MaxFiles),
 		MaxFileSizeMB: dereferenceInt(in.MaxFileSizeMB),
 	}
+	if in.Disabled == nil {
+		// COMPAT(1.7.0): fix for backwards compatibility
+		out.Disabled = !dereferenceBool(in.Enabled)
```

Comment:
If `Enabled = nil` won't this propagate the bug we're trying to fix?

Reply:
Oops, I missed that because I didn't have a test for that case. The API struct gets canonicalized first so we don't hit it in practice, but we should belt-and-suspenders on this!